artichoke-backend/src/extn/core/kernel/mod.rs

```diff
@@ -108,6 +120,57 @@ impl Kernel {
         }
     }
 
+    unsafe extern "C" fn integer(mrb: *mut sys::mrb_state, _slf: sys::mrb_value) -> sys::mrb_value {
+        let interp = unwrap_interpreter!(mrb);
+
+        match integer::Args::extract(&interp) {
+            Ok(args) => match integer::method(&interp, &args) {
+                Ok(v) => v.inner(),
+                Err(integer::Error::InvalidRadix(v, raise_exception)) => {
+                    if raise_exception {
+                        ArgumentError::raisef(interp, "invalid radix", vec![v])
+                    } else {
+                        interp.convert(None::<Value>).inner()
+                    }
+                }
+                Err(integer::Error::InvalidValue(v, raise_exception)) => {
+                    if raise_exception {
+                        ArgumentError::raisef(
+                            interp,
+                            "invalid value for Integer(): \"%S\"",
+                            vec![v],
+                        )
+                    } else {
+                        interp.convert(None::<Value>).inner()
+                    }
+                }
+                Err(integer::Error::ContainsNullByte(raise_exception)) => {
+                    if raise_exception {
+                        ArgumentError::raise(interp, "string contains null byte")
+                    } else {
+                        interp.convert(None::<Value>).inner()
+                    }
+                }
+                _ => RuntimeError::raise(interp, "fatal Kernel#Integer error"),
+            },
+            Err(integer::Error::NoImplicitConversionToString(class_name, raise_exception)) => {
+                if raise_exception {
+                    TypeError::raisef(interp, "can't convert %S into String", vec![class_name])
+                } else {
+                    interp.convert(None::<Value>).inner()
+                }
+            }
+            Err(integer::Error::BaseSpecifiedForNonString(raise_exception)) => {
+                if raise_exception {
+                    ArgumentError::raise(interp, "base specified for non string value")
+                } else {
+                    interp.convert(None::<Value>).inner()
+                }
+            }
+            _ => RuntimeError::raise(interp, "fatal Kernel#Integer error"),
```

Comment:
is this wildcard match arm necessary? What error enum variants are we not handling?